### DISCUSSION TITLE
Log: Always preview diff when commit is selected/focused in list?

### DISCUSSION THREAD

**Discussion by tony** [extrawurst/gitui#1131] [Ideas]
Thank you for the project @extrawurst!

Usability question. Right now the meat of a commit isn't readily viewable (at least in my config as of `0.20.1`)

![image](https://user-images.githubusercontent.com/26336/153053896-810309e0-92a4-449d-b99f-35c7533bf797.png)

Above: for the sake of example only, the bottom right pane is used to show preview. Ultimately opening it anywhere, including a bottom drawer could be possible

I would like to have when select a commit, a pane to show a preview (preferrable scrollable somehow)
![gitui preview mock](https://user-images.githubusercontent.com/26336/153054770-62fbc6b2-caef-4fac-8fe1-8801aef19544.png)

Above: preview without needing to open full commit browser

## logs: gitui 0.20.1 vs tig 2.4.11 diff preview

https://user-images.githubusercontent.com/26336/153051573-334b456b-c1a6-482d-92c6-94bca9bf6ad2.mp4

Above: `tig(1)` vs `gitui(1)` commit viewing behavior upon a single click

https://user-images.githubusercontent.com/26336/153052582-1e60d437-a596-4dac-b6e5-5b9de353ae54.mp4

Above `gitui(1)`: Previewing diff is a few clicks away, each commit seems to be opened separately.  However `gitui(1)`'s opening of a commit's diff is _far_ more intricate and featureful.  I'm speaking of merely having a tig-like preview here upon selection / hover of a commit (not necessarily fully opening the full diff view):

https://user-images.githubusercontent.com/26336/153052158-d0607bdf-9356-4480-bef9-4164ad60546b.mp4

Above (fast paging of previews w/ `tig(1)`): once a preview pane is open in `tig(1)` by pressing "enter" on an individual commit, it persists and pressing up/down will keep the pane open. It acts a lot like mailbird/airmail/geary providing a message preview pane in an email client.

Q: Is this a question or a feature request? If I should rephrase tis as a feature request let me know.

**Comment by extrawurst** [comment]
It is certainly a feature request, I don’t see how to build this without new commands wich sacrifices entry barrier everyone though. the benefit of having this combined total unified diff is not clear to me either. There is still the unclear part of how to scroll it cause it is unlikely it ever fits in view.
